### PR TITLE
Upgrade to new googlecloudexporter

### PR DESCRIPTION
**Pull request by jefferbrecht**
Docs for breaking changes: https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/breaking-changes.md

Summary of changes in this PR:

* Upgrade otel collector to current HEAD (includes 0.49.0 which brings in the new exporter feature flag)
* Change windowsperfcounter objects to maps (this was a 0.49.0 change)
* Enable new exporter via feature flag (Windows + Linux)
* Add three exporter configs for backwards compatibility:
  * `"skip_create_descriptor": true`
  * `"instrumentation_library_labels": false`
  * `"service_resource_labels": false`

### CODE REVIEW

**Comment by jefferbrecht** [comment]
Oops, this is still testing the current exporter. We need to wait until 0.49.0 of the upstream collector is released, and then upgrade the Google collector to point to it, and then point to _that_ from ops-agent, before testing.

**Comment by jefferbrecht** [comment]
@Mrod1598 This PR has some overlap with #534, namely the collector upgrade and changing the windowspercounter objects from strings to maps. This PR seems smaller and is also blocking other activities which need the new exporter so I'd prefer to get this merged first. WDYT?

**Comment by Mrod1598** [comment]
> @Mrod1598 This PR has some overlap with #534, namely the collector upgrade and changing the windowspercounter objects from strings to maps. This PR seems smaller and is also blocking other activities which need the new exporter so I'd prefer to get this merged first. WDYT?
Makes sense to me.

**Comment by jefferbrecht** [comment]
The Debian failure is from the unrelated couchdb installation issue (I remember it passing on Thursday last week). Looks like all other tests pass now.

**Comment by qingling128** [comment]
image

**Comment by qingling128** [comment]
> image

Oh NVM. This is the otel submodule of the operations repo. So pulling from HEAD makes sense.

**Review by qingling128 — APPROVED** [review]
LGTM

File: confgenerator/confgenerator.go [hunk 1]
```diff
@@ -74,9 +74,16 @@ func (uc *UnifiedConfig) GenerateOtelConfig(hostInfo *host.InfoStat) (string, er
...
 					// Receivers are responsible for sending fully-qualified metric names.
-					// NB: If a receiver fails to send a full URL, OT will add the prefix `custom.googleapis.com/opencensus/`.
+					// NB: If a receiver fails to send a full URL, OT will add the prefix `workload.googleapis.com/{metric_name}`.
 					// TODO(b/197129428): Write a test to make sure this doesn't happen.
 					"prefix": "",
+					// OT calls CreateMetricDescriptor by default. Skip because we want
+					// descriptors to be created implicitly with new time series.
+					"skip_create_descriptor": true,
+					// Omit instrumentation labels, which break agent metrics.
+					"instrumentation_library_labels": false,
```
Comment (sophieyfang) [inline confgenerator/confgenerator.go:84]: We will need to sync with @dashpole on this when we start the task on adding instrumentation_library_labels

**Review by sophieyfang — APPROVED** [review]

**Pull request merged**